packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,944 @@
+			break;
+		case TreeShape.Wide:
+			tree = getJSTestTreeWide(numberOfNodes, dataType);
+			break;
+		default:
+			unreachableCase(shape);
+	}
+	return tree;
+}
+
+function getJSTestTreeWide(numberOfNodes: number, dataType: string): Jsonable {
+	const nodes = [];
+	const node = testSubtrees.get(dataType);
+	for (let i = 0; i < numberOfNodes - 1; i++) {
+		nodes.push(node);
+	}
+	const tree = {
+		type: dataSchema.name,
+		fields: {
+			foo: nodes,
+		},
+		value: testSubtrees.get(dataType)?.value,
+	};
+	return tree;
+}
+
+function getJSTestTreeDeep(numberOfNodes: number, dataType: string): Jsonable {
+	const node = testSubtrees.get(dataType);
+	assert(node !== undefined);
+	if (numberOfNodes === 1) {
+		return {
+			type: dataSchema.name,
+			value: node.value,
+		};
+	}
+	const tree = {
+		type: dataSchema.name,
+		fields: {
+			foo: [getJSTestTreeDeep(numberOfNodes - 1, dataType)],
+		},
+		value: node.value,
+	};
+	return tree;
+}
+
+function readTreeAsJSObject(tree: Jsonable, initialTotal: Jsonable, dataType: string): Jsonable {
+	let currentTotal = initialTotal as number | string | boolean | JsonableTree;
+	for (const key of Object.keys(tree)) {
+		if (typeof tree[key] === "object" && tree[key] !== null) {
+			currentTotal = readTreeAsJSObject(tree[key], currentTotal, dataType);
+		}
+		if (key === "value") {
+			assert(tree[key] !== undefined);
+			const currentDataType = typeof tree[key] === "object" ? "Map" : "string";
+			currentTotal = applyOperationDuringRead(currentTotal, tree[key], currentDataType);
+		}
+	}
+	return currentTotal;
+}
+
+/**
+ * changes the value of the leaf node of the Jsonable tree.
+ * @param tree - tree in form of a Jsonable object
+ * @param shape - shape of the tree (wide vs deep)
+ */
+function manipulateTreeAsJSObject(tree: Jsonable, shape: TreeShape): void {
+	let nodesUnderRoot;
+	switch (shape) {
+		case TreeShape.Deep:
+			tree[0].value = 123;
+			break;
+		case TreeShape.Wide:
+			nodesUnderRoot = tree.fields.foo.length;
+			if (nodesUnderRoot === 0) {
+				tree.fields.value = 123;
+			} else {
+				tree.fields.foo[nodesUnderRoot - 1].value = 123;
+			}
+			break;
+		default:
+			unreachableCase(shape);
+	}
+}
+
+function getLeafNodeFromJSObject(tree: Jsonable): Jsonable {
+	for (const key of Object.keys(tree)) {
+		if (typeof tree[key] === "object" && tree[key] !== null) {
+			if (tree[key].type !== undefined && tree[key].fields === undefined) {
+				return tree;
+			}
+			return getLeafNodeFromJSObject(tree[key]);
+		}
+	}
+}
+
+function readCursorTree(
+	forest: IForestSubscription,
+	numberOfNodes: number,
+	shape: TreeShape,
+	dataType: string,
+) {
+	const readCursor = forest.allocateCursor();
+	moveToDetachedField(forest, readCursor);
+	assert(readCursor.firstNode());
+	let nodesRead = 0;
+	let currentTotal = 0 as number | string | boolean | JsonableTree;
+	switch (shape) {
+		case TreeShape.Deep:
+			for (let i = 0; i < numberOfNodes; i++) {
+				assert(readCursor.firstField());
```

Comment:
the editable tree code is walking the tree by looking up the fields by name while this code is taking the first field in iteration order. For your trees this might be the same, but if we want to compare the perf of cursor vs editiable tree, they should walk the trees the same way, so use enterField and enterNode here.

Reply:
updated to use enterField and enterNode